content/library/advanced-features/advanced-widget-behavior.md

```diff
@@ -14,3 +16,252 @@ Widgets are magical and often work how you want. But they can have surprising be
 5. If you don't call a widget function in a script run, we neither store the widget state nor render the widget. If you call a widget function with the same arguments later, Streamlit treats it as a new widget.
 
 4 and 5 are the most likely to be surprising and may pose a problem for some application designs. When you want to persist widget state for recreating a widget, use [Session State](/library/api-reference/session-state) to work around 5.
+
+</Collapse>
+
```

Comment:
This is a very technical and dense page. I almost wonder if we should put a callout at the top to be really clear "this is pretty advanced, in general just use the widgets and don't think about it too much, this page is for debugging or working around common points of confusion"

Not sure if that's a good idea or not, but once we publish this maybe we keep an eye on whether users are getting lost here or not, especially new developers.

Reply:
I can expand that first paragraph a little more to provide some additional context.